### PR TITLE
Enable steps/job for osd ccs

### PR DESCRIPTION
**Pull request by yasun1**
Enable the create/destroy/test steps for the osd ccs cluster:

1. Obtain the credential of osdCcsAdmin
2. Create the osd ccs cluster
3. Create the machine pool on the cluster
4. Configure htpasswd idp on the cluster
6. Destroy the  osd ccs cluster

### New steps

- osd-ccs-osdCcsAdmin-aws
- osd-ccs-cluster-provision-aws
- osd-ccs-conf-machine-pool
- osd-ccs-conf-idp-htpasswd
- osd-ccs-cluster-operators-wait-ready
- osd-ccs-cluster-deprovision
The information (id, console URL, log in, kubeconfig) of the created cluster is shared via the ${SHARED_DIR}.

### workflow

- osd-ccs-e2e-aws creates an advanced osd ccs cluster for the e2e test.

### CODE REVIEW

**Comment by kuiwang02** [comment]
/hold

I am sorry to hold it because I am fear it is merged before I reviewed it. I will try my best to review it and unhold it.
Thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 1]
```diff
@@ -590,6 +590,19 @@ tests:
     test:
     - ref: openshift-extended-test
     workflow: cucushift-installer-rehearse-aws-ipi
+- as: e2e-osd-ccs-nightly-aws
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:593]: Suggest to ask Productivity Team to review. cc @liangxia

File: ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      MACHINE_POOL_INSTANCE_TYPE: "m5.2xlarge"
+      ENABLE_AUTOSCALING: "true"
+      MIN_REPLICAS: "3"
+      MAX_REPLICAS: "6"
+    pre:
+    - ref: osd-ccs-osd-ccs-admin-aws
+    - ref: osd-ccs-cluster-provision-aws
+    - ref: osd-ccs-conf-machine-pool
+    - ref: osd-ccs-conf-idp-htpasswd
+    - ref: osd-ccs-cluster-operators-wait-ready
```
Comment (yunjiang29) [inline ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml:18]: why not create a single chain for provision cluster step? so the user can just focus on the chain  without care about the changes (detailed steps) in the provision step.

File: ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      COMPUTE_MACHINE_TYPE: "m5.xlarge"
+      MULTI_AZ: "true"
+      COMPUTE_NODES: "3"
+      ETCD_ENCRYPTION: "true"
+      MACHINE_POOL_INSTANCE_TYPE: "m5.2xlarge"
+      ENABLE_AUTOSCALING: "true"
+      MIN_REPLICAS: "3"
+      MAX_REPLICAS: "6"
+    pre:
+    - ref: osd-ccs-osd-ccs-admin-aws
```
Comment (yunjiang29) [inline ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml:14]: looks like this step is only generate a new AWS credential, how about rename it to, e.g. `osd-ccs/aws/create-user`, remove the prefix `osd-ccs` under `osd-ccs` folder, and make the step name expressive of its function. The remove function would be `osd-ccs/aws/delete-user`, wdyt?

File: ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    - ref: osd-ccs-cluster-provision-aws
+    - ref: osd-ccs-conf-machine-pool
+    - ref: osd-ccs-conf-idp-htpasswd
+    - ref: osd-ccs-cluster-operators-wait-ready
+    post:
+    - ref: osd-ccs-cluster-deprovision
```
Comment (yunjiang29) [inline ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml:20]: will new created credential be deleted together with destroy command?

**Comment by yunjiang29** [comment]
Hello @yasun1 , I found existing chains `osd-create` and `osd-delete`, to reduce duplicate code, it's better to leverage existing steps/chains,  is it possible to merge your code into existing ones?

File: ci-operator/step-registry/osd-ccs/osd-ccs-admin/aws/osd-ccs-osd-ccs-admin-aws-commands.sh [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+# to get the creadentials.
+# AccessKeyList=($(aws iam list-access-keys --user-name ${username} | jq -r '.AccessKeyMetadata[].AccessKeyId'))
+readarray -t AccessKeyIDList < <(aws iam list-access-keys --user-name ${username} | jq -r '.AccessKeyMetadata[].AccessKeyId')
+if [ "${#AccessKeyIDList[@]}" -ge 2 ]; then
+  echo "Delete the first credential ${AccessKeyIDList[0]}"
+  aws iam delete-access-key --user-name ${username}  --access-key-id ${AccessKeyIDList[0]}
+fi
+echo "Create a new credential..."
+readarray -t AWSToken < <(aws iam create-access-key --user-name ${username} | jq -r '.AccessKey.AccessKeyId,.AccessKey.SecretAccessKey')
+echo "Store the credential in $CREDENTIAL_FILE"
```
Comment (jianlinliu) [inline ci-operator/step-registry/osd-ccs/osd-ccs-admin/aws/osd-ccs-osd-ccs-admin-aws-commands.sh:54]: The user secret is created the on fly when the cluster is created?

**Comment by yasun1** [comment]
After talked with Jianlin and Yunfei, we'd better use the forever credential of osdCcsAdmin but not generate a new one to avoid the potential risk for race condition. This requires
1. Firstly, we need to gather who/which team uses this account to create the osd ccs cluster in its daily works or automation scripts. 
2. Secondly, add the fixed osdCcsAdmin credential into cluster profile, and then update the steps to fit for the changes.

Let us /hold the PR.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 6]
```diff
@@ -590,6 +590,19 @@ tests:
...
+  cron: 0 8 * * 3
+  steps:
+    cluster_profile: aws-qe
+    env:
+      CHANNEL_GROUP: nightly
+      E2E_RUN_TAGS: '@aws-ipi and @osd and @network-openshiftsdn and not @fips'
+      OPENSHIFT_VERSION: "4.11"
+      TAG_VERSION: '@4.11'
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~DisconnectedOnly&;OSD&
```
Comment (kuiwang02) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:602]: @yasun1 as we ever talked, we have OSD and CCS label for golang. for this cluster, which is used? just double confirm it.
how to determine which is used per workflow?

cc @jianzhangbjz

File: ci-operator/step-registry/osd-ccs/cluster/provision/aws/osd-ccs-cluster-provision-aws-commands.sh [hunk 7]
```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+subfix=$(head /dev/urandom | tr -dc a-z0-9 | head -c 4)
+CLUSTER_NAME=${CLUSTER_NAME:-"ci-osd-ccs-$subfix"}
```
Comment (kuiwang02) [inline ci-operator/step-registry/osd-ccs/cluster/provision/aws/osd-ccs-cluster-provision-aws-commands.sh:10]: @yasun1 as we talked, we have two label OSD and CCS. how do we check if it is osd or ccs cluster per this cluster name? thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml (hunk 6 above)
Comment (yasun1) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:602]: The cluster here I provide is an 'osd-ccs' cluster. The 'osd-ccs' means it is an osd cluster created with the customer's cloud account. At here, I missed add `ccs` label.

File: ci-operator/step-registry/osd-ccs/cluster/provision/aws/osd-ccs-cluster-provision-aws-commands.sh (hunk 7 above)
Comment (yasun1) [reply ci-operator/step-registry/osd-ccs/cluster/provision/aws/osd-ccs-cluster-provision-aws-commands.sh:10]: `osd-ccs` is the osd ccs cluster. `osd` will be the osd non-ccs cluster.

**Comment by yasun1** [comment]
There is a blocker card https://issues.redhat.com/browse/SDA-5644 so that now we can not keep a stable credential of  osdCcsAdmin

File: ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml (hunk 3 above)
Comment (yasun1) [reply ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml:14]: As we talked about yesterday, we may close the step.

File: ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml (hunk 4 above)
Comment (yasun1) [reply ci-operator/step-registry/osd-ccs/e2e/aws/osd-ccs-e2e-aws-workflow.yaml:20]: As we talked about yesterday, we may close the step to generate credential.

File: ci-operator/step-registry/osd-ccs/osd-ccs-admin/aws/osd-ccs-osd-ccs-admin-aws-commands.sh (hunk 5 above)
Comment (yasun1) [reply ci-operator/step-registry/osd-ccs/osd-ccs-admin/aws/osd-ccs-osd-ccs-admin-aws-commands.sh:54]: As we talked about yesterday, we may close the step to generate credential.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml (hunk 1 above)
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:593]: Communicate with @yasun1 on slack, they only have 1 or 2 jobs there. We have agreement that it's fine to put it there at the moment. They will migrate to a dedicated job yaml in the future if the job number increase significantly.
Thank you for letting me know @jianlinliu

**Comment by yasun1** [comment]
A dirty PR.

**Pull request closed without merging**